File: adjacentElementsProduct.py
def adjacentElementsProduct(inputArray):
    #return max([inputArray[x]*inputArray[x-1] for x in range(1,len(inputArray))])
    return max(list(map(lambda x, y: x*y, inputArray[0:len(inputArray)-1], inputArray[1:len(inputArray)])))

File: adjacentElementsProduct.java
int adjacentElementsProduct(int[] inputArray) {
    int max=Integer.MIN_VALUE, product=Integer.MIN_VALUE;
    for(int x=0;x<inputArray.length-1;x++){
        product = inputArray[x]*inputArray[x+1];
        if (product>max){
            max = product;
        }
    }
    return max;
}
